sim/hunter/items.go

```diff
@@ -569,6 +570,124 @@ func init() {
 		hunter.ItemSwap.RegisterProc(SignetOfTheUnseenPath, procAura)
 	})
 
+	core.NewItemEffect(StringsOfFate, func(a core.Agent) {
+		hunter := a.(HunterAgent).GetHunter()
+
+		hasMeleeSpecialist := hunter.HasRune(proto.HunterRune_RuneBeltMeleeSpecialist)
+
+		// After using the bow active you cannot gain strands for 30sec.
+		noGainAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:    "Strand Of Fate - No Stack Gain",
+			ActionID: core.ActionID{SpellID: 1231603},
+			Duration: time.Second * 30,
+		})
+
+		// Tracks the number of strands of fate, up to 5 stacks if you're ranged or 4 if you're using the melee version.
+		// TODO: Ranged Hunter's version lets them consume these stacks to move while shooting
+		stacksAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:     "Strand Of Fate - Stacks",
+			ActionID:  core.ActionID{SpellID: 1232946},
+			MaxStacks: core.TernaryInt32(hasMeleeSpecialist, 4, 5),
+			Duration:  time.Second * 20,
+			OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks, newStacks int32) {
+				if hasMeleeSpecialist {
+					strengthChange := float64(20 * (newStacks - oldStacks))
+					hunter.Unit.AddStatDynamic(sim, stats.Strength, strengthChange)
+				}
+			},
+		})
+
+		core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+			Name:       "Strand Of Fate - Ranged Trigger",
+			Callback:   core.CallbackOnSpellHitDealt,
+			Outcome:    core.OutcomeLanded,
+			ProcMask:   core.ProcMaskRanged,
+			ProcChance: 0.1,
+			ExtraCondition: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) bool {
+				return !hasMeleeSpecialist
+			},
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if noGainAura.IsActive() {
+					return
+				}
+				stacksAura.Activate(sim)
+				stacksAura.AddStack(sim)
+			},
+		})
+
+		core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+			Name:       "Strand Of Fate - Melee Trigger",
+			Callback:   core.CallbackOnSpellHitDealt,
+			Outcome:    core.OutcomeLanded,
+			ProcMask:   core.ProcMaskMelee,
+			ProcChance: 0.1,
+			ExtraCondition: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) bool {
+				return hasMeleeSpecialist
+			},
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if noGainAura.IsActive() {
+					return
+				}
+				stacksAura.Activate(sim)
+				stacksAura.AddStack(sim)
+			},
+		})
+
+		// When using the melee version and activating the bow your next strike applies a special Serpent Sting effect.
+		var strands int32
+		SerpentStrikeAura := hunter.GetOrRegisterAura(core.Aura{
```

Comment:
Small knit: capitalization
```suggestion
		serpentStrikeAura := hunter.GetOrRegisterAura(core.Aura{
```